src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfComplexTypesTests.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Spark.Sql;
+using Xunit;
+using static Microsoft.Spark.Sql.Functions;
+
+namespace Microsoft.Spark.E2ETest.UdfTests
+{
+    [Collection("Spark E2E Tests")]
+    public class UdfComplexTypesTests
+    {
+        private readonly SparkSession _spark;
+        private readonly DataFrame _df;
+
+        public UdfComplexTypesTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+            _df = _spark
+                .Read()
+                .Json(Path.Combine($"{TestEnvironment.ResourceDirectory}people_types.json"));
+        }
+
+        /// <summary>
+        /// UDF that takes in Array type.
+        /// </summary>
+        [Fact]
+        public void TestUdfWithArrayType()
+        {
+            // ArrayList works for this type.
+            Func<Column, Column> udfInt = Udf<int[], string>(array => string.Join(',', array));
+
+            Assert.Throws<Exception>(() => _df.Select(udfInt(_df["ages"])).Collect().ToArray());

```

Comment:
This throws an exception without `ToArray()` right? If so, let's remove it.

Reply:
No, actually this won't throw an exception without `ToArray()`

Reply:
OK. then `Show()` works fine? If so, it is not an UDF issue, but deserializing `Row` object.